README.txt
Undocumented configuration


1) Frontend TypoScript

	Property: tx_ncstaticfilecache.disableCache
	Data type: boolean
	Description: By setting this to a true value, the current page and page branch will not be cached statically.
	Example: tx_ncstaticfilecache.disableCache = 1


Changelog
2015-04-16  Tim Lochmüller <tim@fruit-lab.de

	* [!!!] Create a command controller and mark the CLI and Tasks class as deprecated. The classes are removed in a few versions
	* Move the crawler hook to the new (namespace based) location
	* Move the Info module to the new (namespace based) location
	* Migrate the L10N file in the root folder to the new L10N file in Resources/Private/Language/locallang.xml
	* Change extension icon from gif to png
	* Move the main class to the new (namespace based) location
	* Remove the old CLI and the old tasks, because the scheduler do not break directly. Please create the tasks as extbase command controller tasks and change your env

2015-04-15  Tim Lochmüller <tim@fruit-lab.de

	* Remove closing PHP Tags (CGL)
	* Format the PHP classes (CGL)
	* Remove old XCLASS code lines in the footer of the classes
	* Remove $GLOBALS['TYPO3_DB'] to helper method
	* Fix PhpDoc
	* import namespace classes
	* handle GeneralUtility::mkdir_deep in the right way
	* fix undefined variable notice

2015-01-12  Jürgen Kußmann <juergen.kussmann@aoe.com>

	* Add/Update dependency to TYPO3 6.2 and PHP 5.3
	* Use TYPO3-CORE-classes with namespaces

2014-11-11  Klaus Bitto <klaus@netcreators.nl>

	* Release of v2.5.2 to TER.

2014-11-01  Tim Lochmueller  <tim@fruit-lab.de>

	* Fix #61980: Use M code for ExpireByType instead of A (https://forge.typo3.org/issues/61980)
	* Fix #62538: URLs ending with "/" not handled properly (https://forge.typo3.org/issues/62538)

2014-04-10  Klaus Bitto <klaus@netcreators.nl>

	* Release of v2.5.1 to TER.

2014-03-26  Tim Lochmueller  <tim@fruit-lab.de>

	* Remove global statements and use $GLOBAL
	* Smarter explanation for plugins like ExtBase plugins, if the page is not cacheable. More details about extensionName, pluginName...

2014-03-25  Tim Lochmueller  <tim@fruit-lab.de>

	* Wrong severity in call to t3lib_div::devLog()
	* Wrap removed function t3lib_div::intval_positive in the Scheduler task to support 6.x
	* Fix old PHP4 constructor call in the cleaner task
	* Backport changes from TER version to trunk
	* Set dependencies to TYPO3 4.5 minimum
	* Wrap testInt once more, to prefer the TYPO3 CMS 4.7 t3lib_utility_Math function
	* Cleanup and add same classes to the ext_autoload.php
	* Remove PATH_t3lib usage to fit 6.x

2014-03-11  Klaus Bitto  <klaus@netcreators.nl>

	* Fixed bug: Call to undefined method TYPO3\CMS\Core\Utility\GeneralUtility::intval_positive() in class.tx_ncstaticfilecache_tasks_processDirtyPages_AdditionalFieldProvider.php (Thank you, Hendrik Reimers)
	* Increased compatibility to TYPO3 6.2.

2014-03-07  Klaus Bitto  <klaus@netcreators.nl>

	* Integrated TYPO3 6 compatibility adjustments for TER release as v2.4.0. (Selective merge from https://svn.typo3.org/TYPO3v4/Extensions/nc_staticfilecache/trunk/.)

2010-10-13  Oliver Hader  <oliver@typo3.org>

	* Fixed bug: Database elements are removed if directory on the filesystem exists, but could not be accessed

2010-09-20  Oliver Hader  <oliver@typo3.org>

	* Fixed bug: Removing of static files returned wrong boolean value
	* Fixed bug #9850: Small coding errors (thanks to Axel Jung)
	* Raised version to 2.3.3

2010-09-08  Oliver Hader  <oliver@typo3.org>

	* Cleanup: Fixed svn:eol-style of PHP and text files
	* Fixed bug: Only remove database elements if removal in filesystem was successful
	* Added feature: Integrate logging to devLog if clearing static caches failes
	* Raised version to 2.3.2

2010-07-19  Oliver Hader  <oliver@typo3.org>

	* Fixed bug: Typing error in hook name
	* Fixed bug: Infomodule shows creation time instead of last modification

2010-07-15  Oliver Hader  <oliver@typo3.org>

	* Cleanup: Fixed naming and formatting
	* Cleanup: Removed superfluous hook in processDirtyPages() method that was only available in Trunk

2010-07-14  Franz Ripfel  <franz.ripfel@abezet.de>

	* Fixed bug: Clearing cache of a single page deleted also all folders and files of subpages

2010-07-13  Oliver Hader  <oliver@typo3.org>

	* Fixed bug: TYPO3 cache gets cleared on removing expired pages with the markDirtyInsteadOfDeletion setting enabled (thanks to Juergen Kussmann)

2010-05-28  Oliver Hader  <oliver@typo3.org>

	* Added feature: Integrate possibility to disable the clear cache post processing on deman during runtime

2010-05-27  Oliver Hader  <oliver@typo3.org>

	* Fixed bug: markDirtyInsteadOfDeletion property shall only consider specific pages - thus not clear all or pages cache
	* Cleanup: Fixed formatting and inline type hints
	* Fixed bug: Database element is not removed if clearing files did not succeed
	* Fixed bug: Pages with an endtime that would expire a page before the general expiration time is not considered
	* Fixed bug: Additional hash is not written for database elements
	* Fixed bug: Additional hash is not considered for lookups when empty

2010-05-25  Oliver Hader  <oliver@typo3.org>

	* Added feature: Integrate hook to post process the cache scenario after (no matter whether static cache was written)

2010-04-30  Oliver Hader  <oliver@typo3.org>

	* Added feature: Integrate hook to handle deleting a static cached directory

2010-04-19  Oliver Hader  <oliver@typo3.org>

	* Follow-up to bug #5290: Expect the scheme name at first position and allow to modifiy with hook

2010-04-15  Oliver Hader  <oliver@typo3.org>

	* Follow-up to feature of predefining/extending values that are stored in the database
	* Added feature: Add additionalhash to implement individual and more specific database elements (utilized by hooks)
	* Fixed bug: removeExpiredPages triggeres clearing cache of a page multiple times

2010-04-14  Oliver Hader  <oliver@typo3.org>

	* Cleanup: Moved logging part of writing cache files to accordant place
	* Cleanup: Moved information that determine whether a page is cachable and added to them to hook parameters
	* Cleanup: Moved implementation to write compressed content to separate method

ext_emconf.php
<?php

/***************************************************************
 * Extension Manager/Repository config file for ext "nc_staticfilecache".
 *
 * Auto generated 25-03-2014 07:22
 *
 * Manual updates:
 * Only the data in the array - everything else is removed by next
 * writing. "version" and "dependencies" must not be touched!
 ***************************************************************/

$EM_CONF[$_EXTKEY] = array(
	'title'              => 'Static File Cache',
	'description'        => 'Transparent static file cache solution using mod_rewrite and mod_expires. Increase response times for static pages by a factor of 230!',
	'category'           => 'fe',
	'version'            => '3.1.0',
	'state'              => 'stable',
	'modify_tables'      => 'pages',
	'author'             => 'Static File Cache team',
	'author_email'       => 'extensions@netcreators.com',
	'author_company'     => 'Netcreators',
	'constraints'        => array(
		'depends' => array(
			'typo3' => '6.2.0-7.1.99',
		),
	),
);